src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -702,6 +703,7 @@ private void createMainTable() {
                                                          .ifPresent(entry -> {
                                                              preview.setEntry(entry);
                                                              entryEditor.setEntry(entry);
+                                                             setPreviewAction();
```

Comment:
I don't understand this change. Here we react to the user selecting a different entry. For me it is not clear why we need to show or hide the preview in this case (if it already shown, then we don't need to show it; and if it was not shown then it also shouldn't be shown just because the user selects a different entry).

Can you please explain the reason for this change. Thanks!

Reply:
`private void setPreviewAction() {
        PreviewPreferences previewPreferences = Globals.prefs.getPreviewPreferences();
        boolean enabled = previewPreferences.isPreviewPanelEnabled();
        PreviewPreferences newPreviewPreferences = previewPreferences.getBuilder()
                                                                     .withPreviewPanelEnabled(enabled)
                                                                     .build();
        Globals.prefs.storePreviewPreferences(newPreviewPreferences);
        setPreviewActive(enabled);
    }

It gets the current value of whether the preview panel is enabled or disabled. And if enabled the preview is shown. the setPreviewActive(enabled); method call is the part which is responsible for showing the preview and the panel in the UI (previous logic)

` private void setPreviewOn(Boolean previewOn) {
        PreviewPreferences previewPreferences = Globals.prefs.getPreviewPreferences();
        PreviewPreferences newPreviewPreferences = previewPreferences.getBuilder()
                                                                     .withPreviewPanelEnabled(previewOn)
                                                                     .build();
        Globals.prefs.storePreviewPreferences(newPreviewPreferences);

    }`
This method only sets the panel to be enabled. But it's not yet shown in the UI. since setPreviewActive(enabled); is not called in this method.

Thus;

At the beginning by calling setPreviewOn() method enables the preview panel (Not shown in the UI) and the when a row is clicked setPreviewAction() method is called and based on whether the panel is active or not (Panel can be disabled by clicking on the menu item) the preview is shown or not.